synapse/config/server.py

```diff
@@ -641,10 +641,22 @@ def generate_config_section(
             """\
         ## Server ##
 
-        # The domain name of the server, with optional explicit port.
-        # This is used by remote servers to connect to this server,
-        # e.g. matrix.org, localhost:8080, etc.
-        # This is also the last part of your UserID.
+        # The public facing domain of the server
```

Comment:
Overall seems like a big improvement!

```suggestion
        # The public-facing domain of the server
```

To be consistent with the `public_baseurl` below.